src/promptflow/tests/test_configs/flows/chat-with-calorie-assistant/add_message_and_run.py

```diff
@@ -0,0 +1,250 @@
+RUN_STATUS_POLLING_INTERVAL_IN_MILSEC = 1000
+
+
+@tool
+async def add_message_and_run(
+    conn: OpenAIConnection,
+    assistant_id: str,
+    thread_id: str,
+    message: list,
+    assistant_definition: AssistantDefinition,
+    download_images: bool,
+):
+    cli = await get_openai_api_client(conn)
+    invoker = await get_assisant_tool_invoker(assistant_definition)
+    # Check if assistant id is valid. If not, create a new assistant.
+    # Note: tool registration at run creation, rather than at assistant creation.
+    if not assistant_id:
+        assistant = await create_assistant(cli, assistant_definition)
+        assistant_id = assistant.id
+
+    await add_message(cli, message, thread_id)
+
+    run = await start_run(cli, assistant_id, thread_id, assistant_definition, invoker)
+
+    await manage_run(cli, thread_id, invoker, run)
+
+    # await get_run_steps(cli, thread_id, run.id)
+
+    messages = await get_message(cli, thread_id)
+
+    file_id_references = await get_openai_file_references(messages.data[0].content, download_images, conn)
+    return {"content": to_pf_content(messages.data[0].content), "file_id_references": file_id_references}
+
+
+@trace
+async def get_openai_api_client(conn: OpenAIConnection):
+    cli = AsyncOpenAI(api_key=conn.api_key, organization=conn.organization)
+    return cli
+
+
+@trace
+async def get_assisant_tool_invoker(assistant_definition: AssistantDefinition):
+    invoker = AssistantToolInvoker.init(assistant_definition.tools)
+    return invoker
+
+
+@trace
+async def create_assistant(cli: AsyncOpenAI, assistant_definition: AssistantDefinition):
+    assistant = await cli.beta.assistants.create(
+        instructions=assistant_definition.instructions, model=assistant_definition.model
+    )
+    print(f"Created assistant: {assistant.id}")
+    return assistant
+
+
+@trace
+async def add_message(cli: AsyncOpenAI, message: list, thread_id: str):
+    content = extract_text_from_message(message)
+    file_ids = await extract_file_ids_from_message(cli, message)
+    msg = await cli.beta.threads.messages.create(thread_id=thread_id, role="user", content=content, file_ids=file_ids)
+    print("Created message message_id: {msg.id}, assistant_id: {assistant_id}, thread_id: {thread_id}")
+    return msg
+
+
+@trace
+async def start_run(
+    cli: AsyncOpenAI,
+    assistant_id: str,
+    thread_id: str,
+    assistant_definition: AssistantDefinition,
+    invoker: AssistantToolInvoker,
+):
+    tools = invoker.to_openai_tools()
+    run = await cli.beta.threads.runs.create(
+        assistant_id=assistant_id,
+        thread_id=thread_id,
+        model=assistant_definition.model,
+        instructions=assistant_definition.instructions,
+        tools=tools,
+    )
+    print(f"Assistant_id: {assistant_id}, thread_id: {thread_id}, run_id: {run.id}")
+    return run
+
+
+@trace
+async def wait_for_status_check():
+    await asyncio.sleep(RUN_STATUS_POLLING_INTERVAL_IN_MILSEC / 1000.0)
+
+
+@trace
```

Comment:
It's not a good idea to add trace on wait_for_status_check and get_run_status. 

They are too noisy and meaningless.